desk/app/focus.hoon

```diff
@@ -43,62 +59,284 @@
+          :-
+          :~::  poke %goals for a new pool for focus to populate
+            ::
+            %-  ~(poke-our pass:io /pool)
+            :+  %goal-store  %goal-action
+            !>([%4 now.bowl %spawn-pool (crip "focus")])
+            ::  subscribe
+            ::
+            ::   not sure what I get with this?
+            (~(watch-our pass:io /goal-watch) [%goal-store /goals])
+          ==
+          %=  this
+            pool.goals  [our.bowl now.bowl]
+            on.goals  goals.command
+          ==
+      ::
+      ++  check-dependency
+        |=  app=dude:gall
+        ~|  :*  'Your ship is missing the Goals app.'
+                ''
+                'To track goals:'
+                'click on the "Get Urbit Apps" button and search'
+                '~dister-dozzod-niblyx-malnus/gol-cli'
+                ''
+                'Or run the following command in dojo'
+                '|install ~dister-dozzod-niblyx-malnus %gol-cli'
+                ''
+                'If you want to use focus without tracking goals,'
+                'simply use the back button in your browser to leave this screen.'
+            ==
+        ?>  .^(? %gu /(scot %p our.bowl)/[app]/(scot %da now.bowl))
+        ~
+      --
       ::
         %pause
-      ~&  'oops no pause'
-      `this(prev-cmd.state-p %pause)
+      ~&  >  'paused'
+      ::  calc the time left for current timers
+      ::
+      =/  at-pause
+        :-  `@dr`(sub time.then now.bowl)
+            ?:  (lth wrap.then now.bowl)
+              *@dr
+            `@dr`(sub wrap.then now.bowl)
+      :-
+      :~
+        ::  cancel timers
+        ::
+        (~(rest pass:io /(scot %tas mode.state-p)) time.then)
+        (~(rest pass:io /wrap) wrap.then)
+        ::  maybe there is a fin timer
+        ::
+        (~(rest pass:io /fin) time.then)
+        (~(rest pass:io /fin-wrap) wrap.then)
+      ==
+      %=  this
+        left  at-pause
+        prev-cmd.state-p  %pause
+      ==
+      ::
         %cont
-      ~&  'oops all maneuvers'
-      `this(prev-cmd.state-p %cont)
+      ~&  'welcome back'
+      ~&  >>  'continue'
+      =/  time  `@da`(add now.bowl -.left)
+      =/  setwrap  `@da`(add now.bowl +.left)
+      :-
+      ?:  (gte reps reps.groove)
+        :~
+          (~(wait pass:io /fin) time)
+          (~(wait pass:io /fin-wrap) setwrap)
+        ==
+      :~
+        (~(wait pass:io /(scot %tas mode.state-p)) time)
+        (~(wait pass:io /wrap) setwrap)
+      ==
+      %=  this
+        display.state-p  %clock
+        prev-cmd.state-p  %cont
+        then  [time setwrap]
+      ==
       ::
         %public
       `this(public public.command)
-      ::
-        %reveal
-      `this(reveal.state-p +.command)
       ::
         %maneuver
-      ?.  begin.command
-        `this(display.state-p display.command)
-      ~&  'easing in'
-      ~&  "groove be {<gruv.command>}"
-      ::  XX: integrate an actual ease in later
```

Comment:
this ease-in would be nice.
now that goals are in, I want to have a moment to write down the goals first...or at least what I can in either the day or groove title.

I suspect an ease-in time would be best defaulted to zero, but set under an advanced tab. that's probably where Big Rest would be too.